MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesViewModel.java

```diff
@@ -2,43 +2,58 @@
 
 import android.arch.core.util.Function;
 import android.arch.lifecycle.LiveData;
-import android.arch.lifecycle.Transformations;
 import android.arch.lifecycle.ViewModel;
+import android.arch.paging.LivePagedListBuilder;
+import android.arch.paging.PagedList;
 
 import com.novoda.demo.movies.model.Movie;
 import com.novoda.demo.movies.model.Video;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import static android.arch.lifecycle.Transformations.map;
+import static android.arch.lifecycle.Transformations.switchMap;
+
 public class MoviesViewModel extends ViewModel {
 
     private final MovieService movieService;
-    private MoviesSate moviesSate = new MoviesSate(new ArrayList<Movie>(), 1);
+    private LiveData<PagedList<Movie>> paginatedMovies;
+    private LiveData<NetworkStatus> networkStatus;
 
     MoviesViewModel(MovieService movieService) {
         this.movieService = movieService;
+        MoviesDataFactory moviesDataFactory = new MoviesDataFactory(movieService);
+        PagedList.Config pagedListConfig =
+                new PagedList.Config.Builder()
+                        .setEnablePlaceholders(false)
+                        .setPageSize(20)
```

Comment:
The `pageSize` needs to be specified even if the `MoviesApi` doesn't provide a matching query parameter.